src/List.ls

```diff
@@ -6,6 +6,9 @@ each = (f, xs) -->
 map = (f, xs) -->
   [f x for x in xs]
 
+map-ok = (f, xs) -->
+  [res for x in xs when (res = f x)?]
```

Comment:
`for x in xs` is consistent with other functions, but otherwise `[that for xs | f(..)?]` :D